### PR TITLE
feat: 🎸 add a function to get a dataset config's split names

### PR DESCRIPTION
**Pull request by severo**
Also: pass additional arguments (use_auth_token) to get private configs + info of private datasets on the hub

Questions:

- [x] I'm not sure how the versions work: I changed 1.12.1.dev0 to 1.12.1.dev1, was it correct?
    -> no: reverted
- [x] Should I add a section in https://github.com/huggingface/datasets/blob/master/docs/source/load_hub.rst? (there is no section for get_dataset_infos)
    -> yes: added

### CODE REVIEW

**Comment by severo** [comment]
I'm not sure how the versions work: should I change 1.12.1.dev0 to 1.12.1.dev1?

File: setup.py [hunk 1]
```diff
@@ -217,7 +217,7 @@
 
 setup(
     name="datasets",
-    version="1.12.1.dev0",  # expected format is one of x.y.z.dev0, or x.y.z.rc1 or x.y.z (no to dashes, yes to dots)
+    version="1.12.1.dev1",  # expected format is one of x.y.z.dev0, or x.y.z.rc1 or x.y.z (no to dashes, yes to dots)
```
Comment (lhoestq) [inline setup.py:220]: why this change ?

Comment (severo) [reply setup.py:220]: I was not sure if every commit must increase the version or not.

Comment (severo) [reply setup.py:220]: Note that in another project I depend on a commit of the git repository of datasets:

https://github.com/huggingface/datasets-preview-backend/blob/10fd611042b41f2c8bed767a080f3a57b4429653/pyproject.toml#L28

And the dependency is updated only if the version increases. Without this, I have to do:

```
poetry remove datasets
```

add the line manually to pyproject.toml with the new commit, then

```
poetry lock
poetry install
```

Comment (severo) [reply setup.py:220]: I will revert this.

File: src/datasets/inspect.py [hunk 2]
```diff
@@ -129,3 +130,27 @@ def get_dataset_config_names(path: str):
...
+    """
+    builder = load_dataset_builder(path, name=config_name)
+    if builder.info.splits is None:
+        try:
+            return [
+                split_generator.name
+                for split_generator in builder._split_generators(StreamingDownloadManager(base_path=builder.base_path))
+            ]
+        except Exception as err:
+            raise Exception("The split names could not be parsed from the dataset config.") from err
```
Comment (severo) [inline src/datasets/inspect.py:155]: Maybe we should give that Exception its own name? I'm not sure what's the usage in datasets for that case

Comment (lhoestq) [reply src/datasets/inspect.py:286]: Yes feel free to create a new error type for this

**Review by lhoestq — COMMENTED** [review]
Looks good to me ! Can you just merge master into your branch ?
We just deprecated the `script_version` parameter in favor of `revision` in `load_dataset_builder` and `prepare_module`, so you might need to do a few adjustments

**Comment by lhoestq** [comment]
> Should I add a section in https://github.com/huggingface/datasets/blob/master/docs/source/load_hub.rst? (there is no section for get_dataset_infos)

Yes totally :) This tutorial should indeed mention this, given how fundamental it is

File: src/datasets/inspect.py (hunk 2 above)
Comment (severo) [reply src/datasets/inspect.py]: Done

File: src/datasets/inspect.py [hunk 3]
```diff
@@ -99,7 +107,18 @@ def inspect_metric(path: str, local_path: str, download_config: Optional[Downloa
...
-def get_dataset_infos(path: str):
+def get_dataset_infos(
+    path: str,
+    revision: Optional[Union[str, Version]] = None,
+    download_config: Optional[DownloadConfig] = None,
+    download_mode: Optional[GenerateMode] = None,
+    force_local_path: Optional[str] = None,
+    dynamic_modules_path: Optional[str] = None,
+    return_resolved_file_path: bool = False,
+    return_associated_base_path: bool = False,
```
Comment (lhoestq) [inline src/datasets/inspect.py:118]: These parameters may get removed from `prepare_module` in #2986 , so maybe let's not have them here

Comment (severo) [reply src/datasets/inspect.py:118]: ok

Comment (severo) [reply src/datasets/inspect.py]: Done

**Review by lhoestq — APPROVED** [review]
Thanks for the change and for adding it to the documentation :)
LGTM
